File: NeetCode150/Stack/150. Evaluate Reverse Polish Notation/sol.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> s;
        for (const string& token : tokens) {
            if (token == "+" || token == "-" || token == "*" || token == "/") {
                int b = s.top(); s.pop();
                int a = s.top(); s.pop();
                if (token == "+") s.push(a + b);
                if (token == "-") s.push(a - b);
                if (token == "*") s.push(a * b);
                if (token == "/") s.push(a / b);
            } else {
                s.push(stoi(token));
            }
        }
        return s.top();
    }
};

File: NeetCode150/Stack/150. Evaluate Reverse Polish Notation/sol.py
class Solution:
    def is_alnum(self, s):
        return s.isalnum()

    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for i in tokens:
            if i == "+":
                stack.append(stack.pop() + stack.pop())
            elif i == "-":
                a, b = stack.pop(), stack.pop()
                stack.append(b - a)
            elif i == "*":
                stack.append(stack.pop() * stack.pop())
            elif i == "/":
                a, b = stack.pop(), stack.pop()
                stack.append(int(float(b) / a))
            else:
                stack.append(int(i))
        return stack[0]